### PR TITLE
List workspaces

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/11265
Fixes https://github.com/jupyterlab/jupyterlab/issues/10890
Closes https://github.com/jupyterlab/jupyterlab/pull/10923

Reboot of https://github.com/jupyterlab/jupyterlab/pull/10923 using https://github.com/jupyterlab/jupyterlab_server/pull/227

## Code changes

Move workspace management in `jupyterlab_server` in a common class for the handlers and the application to get homogeneous behaviour.

Add `list` subcommand

Remove conditional import of `LicenseApp` as the `jupyterlab_server` version was bumped.

## User-facing changes

New `jupyter lab workspaces list` subcommand

## Backwards-incompatible changes

N/A

### CODE REVIEW

**Comment by fcollonval** [comment]
This needs a release of `jupyterlab_server` with the companion PR as requirement

**Comment by fcollonval** [comment]
Kicking the CI

**Comment by fcollonval** [comment]
Requires https://github.com/jupyterlab/jupyterlab_server/pull/236 to fix workspace export without argument.

**Comment by fcollonval** [comment]
@jtpio could you please make a new release of `jupyterlab_server` with the fix https://github.com/jupyterlab/jupyterlab_server/pull/236?

**Comment by jtpio** [comment]
@fcollonval sure, I'll post here when done

**Comment by jtpio** [comment]
Released: https://github.com/jupyterlab/jupyterlab_server/releases/tag/v2.10.3

**Comment by fcollonval** [comment]
Kicking CI

File: setup.cfg [hunk 1]
```diff
@@ -13,8 +13,8 @@ keywords = ipython, jupyter
 classifiers =
     Development Status :: 5 - Production/Stable
     Framework :: Jupyter
-    # Framework :: Jupyter :: JupyterLab
-    # Framework :: Jupyter :: JupyterLab :: 4
+    Framework :: Jupyter :: JupyterLab
+    Framework :: Jupyter :: JupyterLab :: 3
```
Comment (jtpio) [inline setup.cfg:17]: :+1: 

Probably these were previously commented while waiting for the classifiers to be available.

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.3.x

**Pull request merged**